Add scribeflow plugin

banisterious opened the pull request:
Adding ScribeFlow plugin!

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/banisterious/obsidian-scribeflow

## Release Checklist
- [x] I have tested the plugin on
  - [X]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

banisterious commented:
> #### Hello!
> 
> **I found the following issues in your plugin submission**
> 
> **Errors:**
> 
> ❌ Your repository does not include a license. Go to https://choosealicense.com/ to compare different open source licenses.

Whoops! The file was missing. LICENSE.md is present in the repo now.

banisterious commented:
FYI I have removed the other plugin I am submitting from the JSON (which got there when I forked that PR), as that might have been creating conflicts.

banisterious commented:
Fixes made and released:
1. Moved all inline styles to CSS
2. Renamed command
3. Replaced all innerHTML, outerHTML, and similar API usage with proper DOM API methods
4. Removed all 7 instances of `as any` casting in TemplateIntegrationService.ts

joethei commented:
[--modal-height: 80vh;](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/styles/variables.css#L9-L12)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.

[.btn-base {](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/styles/utilities.css#L8-L307), [.highlight {](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/styles/dashboard/dashboard-table.css#L229-L234)
These CSS classes are a bit to generic, make sure they are unique to your plugin.

[name: 'Create Journal Entry',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/main.ts#L23), [name: 'Open Scribe Dashboard',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/main.ts#L31), ['ScribeFlow Journal Entry'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/main.ts#L38), ['Scribe Dashboard'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/main.ts#L42), [info.createDiv({ text: 'Scribe Dashboard', cls: 'setting-item-name' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L182), [name: 'Sensory Detail',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L72), [name: 'Emotional Recall',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L80), [name: 'Lost Segments',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L88), [name: 'Confidence Score',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L104), [name: 'Character Roles',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L114), [name: 'Characters Count',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L122), [name: 'Familiar Count',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L130), [name: 'Unfamiliar Count',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L138), [name: 'Characters List'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L144), [name: 'Character Clarity',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L147), [name: 'Dream Theme'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L155), [name: 'Symbolic Content'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L156), [name: 'Lucidity Level',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L159), [name: 'Dream Coherence',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L167), [name: 'Environmental Familiarity',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L175), [name: 'Time Distortion',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L183), [name: 'Ease of Recall',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L193), [name: 'Recall Stability',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/types.ts#L201), [return 'Scribe Dashboard';](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L74), [titleSection.createEl('h1', { text: 'Scribe Dashboard' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L135), [{ label: 'Total Entries', value: totalEntries.toString() },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L198), [{ label: 'Average Words', value: avgWords.toString() },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L200-L201), [{ label: 'Daily Goal', value: this.state.statistics.dailyGoalStatus, category: 'goals' },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L992-L994), [label: 'Current Streak',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1011), [label: 'Longest Streak',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1017), [label: 'Days Journaled'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1022), ['Content Insights'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1032), [label: 'With Images',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1035), [label: 'With Dreams',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1041), ['Most Frequent Day'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1050), ['Most Productive Day'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1051), ['Least Productive Day'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1052), [{ label: 'Daily Goal', value: this.state.statistics.dailyGoalStatus, category: 'goals' },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1196-L1198), [{ label: 'Total Entries', value: this.state.statistics.totalEntries, category: 'progress' },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1201-L1202), [label: 'Current Streak',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1211), [label: 'Longest Streak',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1217), [label: 'Days Journaled'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1222), [label: 'Median Words'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1231), [label: 'With Images',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1233), [label: 'With Dreams',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1239), [{ label: 'Most Frequent Day', value: this.state.statistics.mostFrequentDayOfWeek, category: 'pattern' },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1246-L1248). [content.createEl('h4', { text: 'Choose Metrics to Display' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/views/DashboardView.ts#L1372), [const titleText = this.isEditMode ? 'Edit Template' : 'Template Creation Wizard';](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L93), [{ label: 'Creation Method', icon: 'method' },](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L104-L105), [text: 'Choose Creation Method'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L167), ['Direct Input',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L178), ['Template Integration',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L187), ['Predefined Structures',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L196), ['Template Information',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L239), [.setName('Template Name')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L243), [.setName('Description (Optional)')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L256), [text: 'Template Source'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L273), [.setName('Template Plugin')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L297), [.setName(`${pluginDisplayName} Template`)](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L345), [text: 'Template Content'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L435), ['Helper Tools'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L451), [.setButtonText('Insert Sample Entry')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L456), ['Insert Callout'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L459), [.setName('Template Content')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L463), [text: 'Imported Template'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L504), [.setName('Template Content')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L513), ['Templater Conversion'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L532), [name: 'Flat Dual Callout',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L564), [name: '2-Level Nested',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L577), [name: '3-Level Nested',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L591), [text: 'Customize Structure'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L659), [.setName('Template Structure')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L667), ['Update Template' : 'Create Template'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L711), ['Insert Placeholders'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/TemplateWizardModal.ts#L809), ['ScribeFlow Entry'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/JournalEntryModal.ts#L28), [const title = this.contentEl.createEl('h3', { text: 'Journal Structures & Templates' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalStructuresTab.ts#L22), [text: 'Existing Templates'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalStructuresTab.ts#L32), ['Create New Template'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalStructuresTab.ts#L40), [.setName('Create Template')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalStructuresTab.ts#L44), ['Goal Tracking'](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L431), [item.setTitle(`Export Selected (${selectedEntries.length})`)](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/components/ExportContextMenu.ts#L16), [item.setTitle('Export as Plain Text')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/components/ExportContextMenu.ts#L70), [item.setTitle('Export as Plain Text')](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/components/ExportContextMenu.ts#L104), [Copy as Markdown Table',](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/components/ExportButton.ts#L53)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), product / brand names follow the casing set by the respective owner.

[info.createDiv({ text: 'Image settings', cls: 'setting-item-name' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L96), [info.createDiv({ text: 'Image settings', cls: 'setting-item-name' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L71)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

[const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L94-L98), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L180-L185), [foldersContainer.createEl('h6', { text: 'Selected folders:', cls: 'sfp-section-header' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L268), [const goalHeader = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L293-L297), [selectedSection.createEl('h6', { text: 'Selected templates', cls: 'sfp-section-header' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L344), [availableSection.createEl('h6', { text: 'Available templates', cls: 'sfp-section-header' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L371), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L392-L396), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L484-L488), [selectedSection.createEl('h4', { text: 'Selected metrics' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L503), [availableSection.createEl('h4', { text: 'Available metrics' });](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L532), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/settings.ts#L556-L563), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L69-L73), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L153-L157), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L245-L249), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L316-L320), [const goalHeader = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L429-L433), [const heading = containerEl.createDiv('setting-item setting-item-heading');](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/JournalSettingsTab.ts#L531-L538)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[await MarkdownRenderer.renderMarkdown(content, this.contentEl, '', this.plugin);](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/MetricTab.ts#L26), [await MarkdownRenderer.renderMarkdown(content, this.contentEl, '', this.plugin);](https://github.com/banisterious/obsidian-scribeflow/blob/d81c589989cf6e048f342d8d9c733b87e42b6034/src/ui/tabs/InspirationsTab.ts#L102)
Use the non deprecated version of the function, and the component passed to that function is too long-lived, it should only live as long as the content is visible, otherwise you run into memory leaks.  
You can do that by creating a new `Component`, specifying that as the parameter, and making sure that component is unloaded when the view is unloaded.

banisterious commented:
Requested fixes made in [v0.7.3](https://github.com/banisterious/obsidian-scribeflow/releases/tag/0.7.3):
  - CSS Classes: Renamed .btn-base → .scribe-btn-base and .highlight → .scribe-highlight to avoid conflicts
  - Modal Height: Removed global --modal-height: 80vh override that was affecting all Obsidian modals
  - UI Text: Converted all interface text from title case to sentence case (e.g., "Goal Tracking" → "Goal tracking")
  - Settings Headers: Changed "Image settings" to "Images" to remove redundant "settings" suffix
  - Settings API: Replaced custom heading divs with proper new Setting().setHeading() API
  - MarkdownRenderer: Fixed deprecated API usage and implemented proper Component lifecycle to prevent memory leaks
